modules/llrt_tls/src/secure_context.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! SecureContext implementation for TLS
+//!
+//! This module provides the `createSecureContext` function which creates
+//! a reusable TLS context that can be shared between multiple connections.
+
+use std::sync::Arc;
+
+use llrt_utils::object::ObjectExt;
+use rquickjs::{class::Trace, Class, Ctx, Exception, JsLifetime, Object, Result, Value};
+use rustls::pki_types::{pem::PemObject, CertificateDer, PrivateKeyDer};
+use rustls::{ClientConfig, RootCertStore, ServerConfig};
+use webpki_roots::TLS_SERVER_ROOTS;
+
+/// Options for creating a secure context
+#[derive(Default)]
+pub struct SecureContextOptions {
+    pub cert: Option<Vec<u8>>,
+    pub key: Option<Vec<u8>>,
+    pub ca: Option<Vec<Vec<u8>>>,
+    pub ciphers: Option<String>,
+    pub min_version: Option<String>,
+    pub max_version: Option<String>,
+}
```

Comment:
There is a lot of overlap between this and BuildClientConfigOptions, initially I added those options to support TLS options in the fetch API without writing the whole TLS module. Now that we have it, it might make sense to move toward it? We can re-use it in llrt_http.

Reply:
It's fine to do it after, thoughts @richarddavison